pkg/api/submit.proto

```diff
@@ -26,6 +26,47 @@ message JobSubmitRequestItem {
     // Indicates which scheduler should manage this job.
     // If empty, the default scheduler is used.
     string scheduler = 11;
+    // Main object that determines when an application has finished.
+    KubernetesMainObject mainObject = 12;
+    // Set of additional Kubernetes objects to create as part of the job.
+    repeated KubernetesObject objects = 13;
+}
+
+// Kubernetes objects that can serve as main objects for an Armada job.
+message KubernetesMainObject {
+    ObjectMeta objectMeta = 1;
+    oneof object {
```

Comment:
Can we just remove the oneof here since its only a single field?

Reply:
No, it's necessary to support future extensions.